aspnetcore/release-notes/aspnetcore-6.0.md

```diff
@@ -0,0 +1,514 @@
+
+### HTTP logging middleware
+
+HTTP logging is a new built-in middleware that logs information about HTTP requests and HTTP responses including the headers and entire body:
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_httplg&highlight=4)]
+
+Navigating to `/` with the previous code logs infomation similar to the following:
+
+```dotnetcli
+info: Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware[1]
+      Request:
+      Protocol: HTTP/2
+      Method: GET
+      Scheme: https
+      PathBase: 
+      Path: /
+      Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
+      Accept-Encoding: gzip, deflate, br
+      Accept-Language: en-US,en;q=0.9
+      Cache-Control: max-age=0
+      Connection: close
+      Cookie: [Redacted]
+      Host: localhost:44372
+      User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36 Edg/95.0.1020.30
+      sec-ch-ua: [Redacted]
+      sec-ch-ua-mobile: [Redacted]
+      sec-ch-ua-platform: [Redacted]
+      upgrade-insecure-requests: [Redacted]
+      sec-fetch-site: [Redacted]
+      sec-fetch-mode: [Redacted]
+      sec-fetch-user: [Redacted]
+      sec-fetch-dest: [Redacted]
+info: Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware[2]
+      Response:
+      StatusCode: 200
+      Content-Type: text/plain; charset=utf-8
+```
+
+The preceding output was enabled with the following *appsettings.development.json* file:
+
+[!code-json[](aspnetcore-6.0/samples/WebApp1/appsettings.development.json?highlight=6)]
+
+HTTP logging provides logs of:
+
+* HTTP Request information
+* Common properties
+* Headers
+* Body
+* HTTP Response information
+
+To configure the HTTP logging middleware, specify <xref:Microsoft.AspNetCore.HttpLogging.HttpLoggingOptions>:
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_httplg2&highlight=1,4-13)]
+
+### IConnectionSocketFeature
+
+The <xref:Microsoft.AspNetCore.Connections.Features.IConnectionSocketFeature> request feature provides access to the underlying accept socket associated with the current request. It can be accessed via the <xref:Microsoft.AspNetCore.Http.Features.FeatureCollection> on `HttpContext`.
+
+For example, the following app sets the <xref:System.Net.Sockets.Socket.LingerState> property on the accepted socket:
+
+<!-- TODO FIX code -->
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_icsf)]
+
+### .NET Hot Reload
+
+<!--TODO @LyalinDotCom this should probably be removed and left to the .NET 6 what's new doc -->
+Hot Reload minimizes the number of app restarts after code changes. For more information, see [Update on .NET Hot Reload progress and Visual Studio 2022 Highlights](https://devblogs.microsoft.com/dotnet/update-on-net-hot-reload-progress-and-visual-studio-2022-highlights/)
```

Comment:
@LyalinDotCom this should probably be removed and left to the .NET 6 what's new doc